docs/tutorials/quesion-answering-bot.md

```diff
@@ -0,0 +1,224 @@
+
+One way to solve this is by predicting answers using a neural network that was trained on pairs of quesitons and their corresponding answers. In many cases such a dataset is not available, like in the case of most software documentation. Let's say we want to build a chatbot to answer questions about the jina documentation. What if I told you that there is a way to reframe this task as a search problem and that this would alleviate the need for a large dataset of matching questions and answers?
+
+How you ask? *Let me explain!*
+
+## Overview 
+Our approach to the problem leverages the [Doc2query method](https://arxiv.org/pdf/1904.08375.pdf), which, form a piece of text, predicts different questions the text could potentially answer. For example, given a sentence such as `Jina is an open source framework for neural search.`, the model predicts questions such as `What is jina?` or `Is jina open source?`.
+
+The idea here is to predict a number of questions for every part of the original text document, in our case the jina documentation. Then we use an encoder to create a vector representation for each of the predicted questions. These representations are stored and provide the index for our body of text. When a user prompts the bot with a question, we encode it in the same way we encoded our generated questions. Now we can run a similarity search on the encodings. The encoding of the user's query is compared with the encodings in our index to find the closes match.
+
+Since we know what part of the original text was used to generate the question, that was most similar to the user's query, we can return the original text as an answer to the user.
+
+Now that you have a general idea of what we will be doing, the following section will show you how to define our `Flow`s in jina. Then we will take a look at how to implement the necessary `Executor`s for our search-based question-answering system.  
+
+## Indexing the text document 
+Let's imagine we extracted a bunch of sentences from jina's documentation and stored them in a `DocumentArray`, as shown below. 
+
+```python
+example_sentences = [
+    "Document is the basic data type that Jina operates with",
+    "Executor processes a DocumentArray in-place", 
+    ...,
+    "Jina uses the concept of a flow to tie different executors together"
+]
+
+docs = DocumentArray([Document(content=sentence) for sentence in example_sentences])
+```
+
+As described in the last section, we first need to predict potential questions for each of the elements in the `DocumentArray`. Then we have to use another model to create vector encodings from the the predicted questions. Finally, we store them as the index. 
+
+At this point we have enough information to start defining our `Flows`.
+
+*Without further ado, let's build!*
+
+``` python
+indexing_flow = Flow(
+# Generate potential questions using doc2query
+).add(name="question_transformer", 
```

Comment:
can u keep single quotes as it is more aligned with our style

Reply:
That's a good point! Changing it!